src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,29 @@
+import { useState } from 'react';
+import { Product } from '../types/Product';
+
+export function useLocalStorage(
+  key: string, defoultItems: Product[],
+): [Product[], (i: Product[]) => void] {
```

Comment:
```suggestion
export function useLocalStorage<T>(
  key: string, defoultItems: T,
): [T, (i: T) => void] {
```
it's better to make it more reusable (use generic type) to allow saving anything, or rename the hook accordingly to show we save products only in the local storage